### PR TITLE
feat: refactor architecture

### PR DESCRIPTION
**Pull request by 0xpanoramix**
## Description

This PR provides a refactor of the project architecture.

## Changes include

- [ ] Bugfix (non-breaking change that solves an issue).
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention).
- [x] New feature (non-breaking change that adds functionality).
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality).
- [ ] Documentation update (change added to any kind of documentation).

## Checklist

- [x] I have assigned this PR to myself.
- [x] I have added at least 1 reviewer.
- [x] I have added the relevant labels.
- [x] I have updated the official documentation.
- [ ] I have added sufficient documentation in code.

## Testing

- [ ] I have tested this code with the official test suite.
- [x] I have tested this code manually.

### CODE REVIEW

File: .envrc.example
Comment (Cleopha) [inline .envrc.example:1]: does not write a hard the DB url uses the other env variable for the user, password, port, db_name etc...

File: .eslintrc.yml
Comment (Cleopha) [inline .eslintrc.yml:1]: why yml not json ?

File: backend/app.session.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    interface SessionData {
+        user: UserSession;
+    }
+}
+
+export default session({
+    secret: ServerConfiguration.sessionSecret,
+    resave: false,
+    saveUninitialized: false,
+    cookie: { maxAge: 1000 * 60 * 60 * 24 * 7 },
```
Comment (Cleopha) [inline backend/app.session.ts:20]: put a comment to know how much it makes in hour or day

File: backend/app.types.ts
Comment (Cleopha) [inline backend/app.types.ts:1]:
```ts
export interface ErrorRO {
    readonly statusCode: StatusCodes;
    readonly message: string;
}
```
I guess you can use `readonly` keyword

File: backend/database/schemas/entities.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+export interface PgmigrationEntity {
+  id: number
+  name: string
+  run_on: Date
+}
+
+export interface SessionEntity {
+  created_at: Date
+  expire: Date | null
+  sess: any
```
Comment (Cleopha) [inline backend/database/schemas/entities.ts:10]: oO an "any" type

**Review by Cleopha — COMMENTED** [review]
some light commentary.
Otherwise LGTM 🚀

File: .eslintrc.yml
Comment (0xpanoramix) [reply .eslintrc.yml:1]: I always used YAML configurations for eslint.

File: backend/app.types.ts
Comment (0xpanoramix) [reply backend/app.types.ts:1]: No, this will be modified by the error middleware.

File: backend/database/schemas/entities.ts (hunk 2 above)
Comment (0xpanoramix) [reply backend/database/schemas/entities.ts:10]: This is an auto-generated file.

File: backend/app.session.ts (hunk 1 above)
Comment (0xpanoramix) [reply backend/app.session.ts]: Fixed in https://github.com/csc-667-spring-2023-roberts/term-project-frenchies/pull/3/commits/a478fa53114858fec3541e4126fdd9e4b23cd530

File: .envrc.example
Comment (0xpanoramix) [reply .envrc.example:1]: Fixed in https://github.com/csc-667-spring-2023-roberts/term-project-frenchies/pull/3/commits/a478fa53114858fec3541e4126fdd9e4b23cd530

**Review by Cleopha — APPROVED** [review]

**Review by antoinerosselli — APPROVED** [review]

**Review by FlorianCartozo — APPROVED** [review]

**Pull request merged**